tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,941 @@
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(
+            'searchengine' => array(
+                'solr' => array(
+                    'default' => array(
+                        'service' => array(
+                            'index' => array(
+	                            'endpoint' => array(
+		                            'primary' => array(
+			                            'host' => 'examplehost',
+			                            'port' => 'exampleport',
+			                            'path' => 'this_solr_instance_name_does_not_exist'
+		                            )
+	                            )
+                            )
+                        )
+                    )
+                )
+            )
+        ) );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	public function testPrepareAndOutputXML() {
+		$this->adjustConfiguration( array(
+			                            'log' => array( 'prepare' => array( 'xml' => true ) )
+		                            ) );
+
+		$this->_addOneDocumentToIndex();
+	}
+
+	public function testAddDocumentToEmptyIndex() {
+		$this->_addOneDocumentToIndex();
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testRemoveDocumentFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+		$document = new Opus_Document( $this->document_id );
+		$this->indexer->removeDocumentsFromIndex( $document );
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testRemoveDocumentArrayFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+		$document = new Opus_Document( $this->document_id );
+		$this->indexer->removeDocumentsFromIndex( array( $document ) );
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testRemoveDocumentByIdFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+		$this->indexer->removeDocumentsFromIndexById( $this->document_id );
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	public function testRemoveDocumentArrayByIdFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->assertEquals( 1, $this->_getNumberOfIndexDocs() );
+
+		$this->indexer->removeDocumentsFromIndexById( array( $this->document_id ) );
+
+		$this->assertEquals( 0, $this->_getNumberOfIndexDocs() );
+	}
+
+	/**
+	 * @expectedException \InvalidArgumentException
+	 */
+	public function testRemoveNullFromIndex() {
+		$this->_addOneDocumentToIndex();
+
+		$this->indexer->removeDocumentsFromIndex( null );
+	}
+
+	/**
+	 * @expectedException \InvalidArgumentException
+	 */
+	public function testRemoveMissingDocumentFromIndex() {
```

Comment:
That's a warning ... I prepended call with @ to suppress it. Is there any better option?